tools/makemanifest.py: Allow passing flags to mpy-tool.py

Jongy opened the pull request:
Use it to remain with the default MPZ_DIG_SIZE = 32 on 64-bit builds.

Jongy committed:
tools/makemanifest.py: Allow passing flags to mpy-tool.py.

dpgeorge commented on ports/unix/Makefile:
```diff
@@ -178,7 +178,9 @@ ifneq ($(FROZEN_MANIFEST)$(FROZEN_MPY_DIR),)
 CFLAGS += -DMICROPY_QSTR_EXTRA_POOL=mp_qstr_frozen_const_pool
 CFLAGS += -DMICROPY_MODULE_FROZEN_MPY
-CFLAGS += -DMPZ_DIG_SIZE=16 # force 16 bits to work on both 32 and 64 bit archs
+ifneq ($(MICROPY_FORCE_32BIT),1)
```
I'm not sure this will catch all build scenarios: eg what about building on a 32-bit machine?  Then this config option may not be set but the dig size should still be 16.

Jongy replied:
Hmm, you're right. I'll think of another way to check the compiler version perhaps.

Without this change, since mpy-tool assumes MPZ_DIG_SIZE=16, you must define MPZ_DIG_SIZE=16 in your CFLAGS/port to have it compile. It better be automatic.

dpgeorge replied:
If you rebase this PR on to latest master that will include a new Travis build for 32-bit unix, which can test this.

dpgeorge replied:
I think we should just remove this change.

Jongy commented:
I wonder if it's required, after all - I mean, I added this because my first idea was to change mpy-tool to use digit size = 32, instead of forcing the Linux kernel port to use digit size = 16.

Jongy commented:
Sigh, github hotkeys.

dpgeorge commented:
@Jongy can you please rebase on latest master so it runs the CI (there are a lot of new tests now).  Then I can comment on the above.

Jongy commented:
> @Jongy can you please rebase on latest master so it runs the CI (there are a lot of new tests now). Then I can comment on the above.

Done @dpgeorge . I will take another look at what I did here to recall what required it. It was something in ##5482.

dpgeorge commented:
This is erroring on qemu_mips and qeum_arm builds with:
```
CC build-coverage/frozen_content.c
CC coverage.c
build-coverage/frozen_content.c:38:2: error: #error "incompatible MPZ_DIG_SIZE"
   38 | #error "incompatible MPZ_DIG_SIZE"
      |  ^~~~~
make: *** [../../py/mkrules.mk:77: build-coverage/build-coverage/frozen_content.o] Error 1
```

dpgeorge commented on tools/makemanifest.py:
```diff
@@ -248,6 +248,7 @@ def main():
         "-f", "--mpy-cross-flags", default="", help="flags to pass to mpy-cross"
     )
     cmd_parser.add_argument("-v", "--var", action="append", help="variables to substitute")
+    cmd_parser.add_argument('--mpy-tool-flags', default='', help='flags to pass to mpy-tool')
```
please use double quotes `"` (to satisfy Black formatting)

dpgeorge commented:
> I think we can keep the mpy-tool addition, but remove the change in the unix port.

Agreed.

Jongy commented:
@dpgeorge , I have:
1. Removed the unix port change, as discussed
2. Simplified the call to `makemanifest.py` - it's okay to call with empty `MPY_TOOL_FLAGS` because `args.mpy_tool_flags.split()` will evaluate to `[]`.

and fixed other CI notes

dpgeorge commented:
Thank you!
